samples/client/petstore/k6/script.js

```diff
@@ -23,182 +23,335 @@ let apiKey = "TODO_EDIT_THE_API_KEY";
-        let request = http.get(url);
-        check(request, {
+        {
+          let url = BASE_URL + `/store/order`;
+          // TODO: edit the parameters of the request body.
+          let body = {"id": "long", "petId": "long", "quantity": "integer", "shipDate": "date", "status": "string", "complete": "boolean"};
+          
+          let params = {headers: {"Content-Type": "application/json", "Accept": "application/json"}};
+          
+          
+          let request = http.post(url, JSON.stringify(body), params);
+          
+          check(request, {
             "successful operation": (r) => r.status === 200
-        });
-        sleep(SLEEP_DURATION);
+          });
+          
+          
+        }
+        
     });
     group("/user/logout", () => {
-        let url = BASE_URL + `/user/logout`;
+        
         // Request No. 1
-        let request = http.get(url);
-        check(request, {
+        {
+          let url = BASE_URL + `/user/logout`;
+          
+          
+          
+          let request = http.get(url);
+          
+          check(request, {
             "successful operation": (r) => r.status === 200
-        });
-        sleep(SLEEP_DURATION);
+          });
+          
+          
+        }
+        
     });
-    group("/user/{username}", () => {
-        let username = 'TODO_EDIT_THE_USERNAME'; // specify value as there is no example value for this parameter in OpenAPI spec
-        let url = BASE_URL + `/user/${username}`;
+    group("/pet/{petId}", () => {
+        let petId = 'TODO_EDIT_THE_PETID'; // specify value as there is no example value for this parameter in OpenAPI spec
+        
         // Request No. 1
-        let request = http.get(url);
-        check(request, {
+        {
+          let url = BASE_URL + `/pet/${petId}`;
+          
+          
+          
+          let request = http.get(url);
+          
+          check(request, {
             "successful operation": (r) => r.status === 200
-        });
-        sleep(SLEEP_DURATION);
-
+          });
+          
+          
+           sleep(SLEEP_DURATION);
+        }
+        
+        
         // Request No. 2
-        // TODO: edit the parameters of the request body.
-        body = {"id": "long", "username": "string", "firstName": "string", "lastName": "string", "email": "string", "password": "string", "phone": "string", "userStatus": "integer"};
-        params = {headers: {"Content-Type": "application/json"}};
-        request = http.put(url, body, params);
-        sleep(SLEEP_DURATION);
-
-        // Request No. 3
-        request = http.del(url);
-        sleep(SLEEP_DURATION);
+        {
+          let url = BASE_URL + `/pet/${petId}`;
+          
+          let params = {headers: {"api_key": `${apiKey}`, "Accept": "application/json"}};
+          
+          // this is a DELETE method request - if params are also set, empty body must be passed
+          let request = http.del(url, {} , params);
```

Comment:
I know of this issue and this is probably a trade-off between readability of the template and the intended results.
@wing328 Can you advise us on this particular issue? You definitely have more experience than both of us in this regard.

Reply:
You need to reformat the template so it looks ugly for us. In the mustache file on line 41 you will have something like:

```
{{/params}}{{^params}}
```

which is kinda equivalent to one line `} else if {` your current mustache formatting currently is equivalent to:
```
} else
     if   
      {
```

Reply:
You need to merge the mustache `not` tags into closing of previous tag at least.

Reply:
Hi, @mostafa and @agilob

Unfortunately, what @agilob suggested does not work as expected; what's odd is that it results in additional new line, instead of less new lines.

I checked several different combinations before I replied few days ago (2021/11/02). Also, while working on this few weeks ago. None of them allowed to keep the mustache template readable AND have less whitespace in the generated artifact - yes, it's a balance between readability of the template and the resulting generated artifact, as @mostafa mentioned.

I also checked the code for Mustache templating engine (`com.samskivert.mustache.Mustache`) used by `org.openapitools.codegen.DefaultCodegen` (`org.openapitools.codegen.DefaultCodegen.templatingEngine`) to check if perhaps there's some setting we can use but did not find anything.

Please know this is a generated artifact which is ready to run, as per documentation, example project and presentation I gave during recent EclipseCon 2021 (see links in this thread, posted earlier). It does not need to be touched by hand - it can be re-generated as needed. However, IF there is need to modify it by hand, then simply opening the generated artifact in IDE which has built-in support for Mustache and running a built-in formatter will automatically make the extra white space go away.

I'm sorry I cannot help more on this.

Reply:
P.S. See "whitespace control character" (`~`) mentioned in third section @ https://zordius.github.io/HandlebarsCookbook/0031-whitespace-control.html - this would be the easiest, but it DOES NOT work in the Mustache templating engine used currently in this project. I did check this and wanted to use this, that is how I know.

Reply:
@ideas-into-software Fair enough!

@wing328 I suggest we merge it as is.

Reply:
>@wing328 I suggest we merge it as is.

No... let's not merge code that can produce 20 empty lines in a row or 2 lines in a row with 8 spaces in them, really bad idea for tracibility.

@ideas-into-software I did exactly what I explained above and opened PR to your branch: https://github.com/DataInMotion/openapi-generator/pull/1

Reply:
Hi @agilob

I reviewed your PR (https://github.com/DataInMotion/openapi-generator/pull/1), testing all this locally. Please take into account the following: 

 **1**. I explicitly mentioned - what @mostafa also understands - that the mustache template becomes unreadable when all new lines between different kinds of tags is removed; what you did in your PR I already did while ago when working on this, but the result is what I mentioned earlier few times. 
 
  What you mentioned in your earlier comment is "You need to merge the mustache not (^check) tags into closing of previous tag at least." - that's what I was referring to in earlier comment, i.e. that this by itself does not change anything in terms of extra whitespace. Whitespace control character would be the solution, but this mustache processor does not support it (see my earlier comment).
  

 **2**. I noticed you or someone else made changes to the `K6ClientCodegen` which is part of this PR you made - there are several issues with these changes:

 **a**) @ #`preprocessOpenAPI` method 

 The original intent for swallowing that exception is that not all operations have parameters and there's absolutely nothing wrong with that - so if `NullPointerException` is thrown we can safely ignore it, instead of logging it, like you started logging it via `LOGGER.error("{}", e.getMessage(), e);`; this results in unecessary stack traces printed out in output when running the generator.


 **b**) @ #`shouldHideOperationResponse` method

 Intent for using switch / case is additional extensions / extension params which need to be handled differently (see `getDataExtractSubstituteParameter` method for comparison); for some reason someone removed that switch.
 
 
 **c**) @ #`calculateRequestOrder` method

 Signature of this method mentions Integer, but your refactor returns primitive int, i.e. autoboxing kicks in, which is a performance issue.
 

I'll gladly merge this PR you made but these would need to be addressed.

Reply:
>1. I explicitly mentioned - what @mostafa also understands - that the mustache template becomes unreadable when all new lines between different kinds of tags is removed; what you did in your PR I already did while ago when working on this, but the result is what I mentioned earlier few times.


Because this is how we develop templates, to produce generated code that is easy to read and understand. Generated code must be easy to follow, that's a profit here, but we're paying the price of having complex mustache templates. Check complexity of [this template](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/dart/libraries/dio/api.mustache#L45) if you don't believe.

> this results in unecessary stack traces printed out in output when running the generator.

The stacktraces are ABSOLUTELY necessary because they point out problems with the generator or configuration. You can change it to remove 3rd param from logging, but please replace logger to say which field is failing and why. I didn't have time to add this info, looks like you know more about this generator to fix the NPE problem correctly.

>Signature of this method mentions Integer, but your refactor returns primitive int, i.e. autoboxing kicks in, which is a performance issue.

There is no performance issue. I replaced two `Integer.valueOf()` with _one_ autoboxing on return statement. I put two `int`s in place of two `Integer`s. Technically this is faster, but was not the point, it's simply less verbose to read that return statement. There are much bigger problems in openapi in performance and memory leaks.

>I'll gladly merge this PR you made but these would need to be addressed.

You said above:

>Unfortunately, what @agilob suggested does not work as expected; what's odd is that it results in additional new line, instead of less new lines.

This PR is take it or leave it. I showed you how to write the template to prevent getting 4 empty lines every 2 line. It can't be merged in current state where we don't know what is throwing NPE and when generated code will add +300 empty lines to my project.

Reply:
@agilob Thanks for the comments! I think it's nothing to worry about, as the issues we mentioned might be fixed. 🙂 

@ideas-into-software Thanks for your contribution! 🙏  I'll take care of the rest to make this PR mergeable.

Reply:
thanks, I'll be happy to test it on my project :+1: